### PR TITLE
VideoCommon: move hirestexture data to reusable class called CustomTextureData

### PR DESCRIPTION
**Pull request by iwubcode**
Split out from #11300 and replaced existing VideoCommon code with `CustomTextureData`

### CODE REVIEW

File: Source/Core/VideoCommon/GraphicsModSystem/Runtime/CustomTextureData.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    u32 height = 0;
+    u32 row_length = 0;
+  };
+  std::vector<Level> m_levels;
+};
+
+bool LoadDDSTexture(CustomTextureData* texture, const std::string& filename);
+bool LoadDDSTexture(CustomTextureData::Level* level, const std::string& filename, u32 mip_level);
+bool LoadPNGTexture(CustomTextureData* texture, const std::string& filename);
+bool LoadPNGTexture(CustomTextureData::Level* level, const std::string& filename);
```
Comment (AdmiralCurtiss) [inline Source/Core/VideoCommon/GraphicsModSystem/Runtime/CustomTextureData.h:29]: Please put this in some namespace.

File: Source/Core/VideoCommon/HiresTextures.h [hunk 2]
```diff
@@ -17,7 +18,7 @@ enum class TextureFormat;
 std::set<std::string> GetTextureDirectoriesWithGameId(const std::string& root_directory,
                                                       const std::string& game_id);
 
-class HiresTexture
+class HiresTexture final : public CustomTextureData
```
Comment (AdmiralCurtiss) [inline Source/Core/VideoCommon/HiresTextures.h:21]: Can you instead of the inheritance use a CustomTextureData member in this class?

**Comment by AdmiralCurtiss** [comment]
Most of the functions in `CustomTextureData.cpp` also look like they could be `static`.

File: Source/Core/VideoCommon/HiresTextures.h [hunk 3]
```diff
@@ -29,31 +30,20 @@ class HiresTexture
...
   static std::unique_ptr<HiresTexture> Load(const std::string& base_filename, u32 width,
                                             u32 height);
-  static bool LoadDDSTexture(HiresTexture* tex, const std::string& filename);
-  static bool LoadDDSTexture(Level& level, const std::string& filename, u32 mip_level);
-  static bool LoadTexture(Level& level, const std::vector<u8>& buffer);
   static void Prefetch();
 
   HiresTexture() = default;
   bool m_has_arbitrary_mipmaps = false;
+  VideoCommon::CustomTextureData m_data;
```
Comment (AdmiralCurtiss) [inline Source/Core/VideoCommon/HiresTextures.h:48]: While you're here... can you swap these two members (so the memory layout is the same as before, probably doesn't matter but y'know) and put an empty line between the constructor and data?

Comment (iwubcode) [reply Source/Core/VideoCommon/HiresTextures.h:48]: Whoops, apologies.  Fixing...

**Review by AdmiralCurtiss — APPROVED** [review]
Seems fine. Untested.

**Review by lioncash — APPROVED** [review]

**Pull request merged**